### PR TITLE
fix: allow to configure internal and public node urls separately

### PR DESCRIPTION
**Pull request by spetrunin**
#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by jensneuse — APPROVED** [review]

File: packages/sdk/src/configure/index.ts [hunk 1]
```diff
@@ -817,17 +817,17 @@ export const configureWunderGraphApplication = (config: WunderGraphConfigApplica
 
 			done();
 
-			let nodeUrl = resolveConfigurationVariable(resolved.nodeOptions.nodeUrl);
+			let publicNodeUrl = resolveConfigurationVariable(resolved.nodeOptions.publicNodeUrl);
 			// trim trailing slash if any
-			nodeUrl = nodeUrl.endsWith('/') ? nodeUrl.slice(0, -1) : nodeUrl;
```
Comment (StarpTech) [inline packages/sdk/src/configure/index.ts:822]: Could you put this in a helper fn?

File: packages/sdk/src/configure/options.ts [hunk 2]
```diff
@@ -99,10 +103,15 @@ export const resolveNodeOptions = (options?: NodeOptions): ResolvedNodeOptions =
 		? new EnvironmentVariable(WgEnv.NodeUrl, `http://localhost:${options?.listen?.port}`)
 		: DefaultNodeOptions.nodeUrl;
 
+	const fallbackPublicNodeUrl = options?.listen?.port
```
Comment (StarpTech) [inline packages/sdk/src/configure/options.ts:106]: Why is the `port` significant for the condition? I'd say the whole URL is relevant.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: packages/sdk/src/configure/options.ts [hunk 3]
```diff
@@ -94,15 +102,19 @@ export interface ResolvedServerLogger {
...
+export const resolveNodeOptions = (options?: NodeOptions): ResolvedNodeOptions => {
 	let nodeOptions = isCloud
 		? DefaultNodeOptions
 		: {
-				nodeUrl: options?.nodeUrl || fallbackNodeUrl,
+				nodeUrl:
+					options?.nodeUrl || new EnvironmentVariable(WgEnv.NodeUrl, fallbackUrl(defaultNodePort, options?.listen)),
+				publicNodeUrl:
+					options?.publicNodeUrl ||
+					new EnvironmentVariable(WgEnv.PublicNodeUrl, fallbackUrl(defaultNodePort, options?.listen)),
```
Comment (StarpTech) [inline packages/sdk/src/configure/options.ts:117]: Shouldn't it be?

```ts
const nodeUrl = options?.nodeUrl || new EnvironmentVariable(WgEnv.NodeUrl, fallbackUrl(defaultNodePort, options?.listen))
const publicNodeUrl = options?.publicNodeUrl || nodeUrl,
```

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: packages/sdk/src/configure/options.ts (hunk 3 above)
Comment (spetrunin) [reply packages/sdk/src/configure/options.ts:117]: than potentially it will share the same env var, which is not desired I believe

Comment (StarpTech) [reply packages/sdk/src/configure/options.ts:117]: Make sense, thanks.

**Review by StarpTech — APPROVED** [review]
LGTM

File: pkg/node/node.go [hunk 4]
```diff
@@ -416,7 +416,7 @@ func (n *Node) startServer(nodeConfig WunderNodeConfig) error {
 	}
 
 	n.log.Debug("node url",
-		abstractlogger.String("nodeUrl", nodeConfig.Api.Options.NodeUrl),
+		abstractlogger.String("nodeUrl", nodeConfig.Api.Options.PublicNodeUrl),
```
Comment (StarpTech) [inline pkg/node/node.go:419]: Does it make sense to print here a URL that is not controlled by the node?

**Review by StarpTech — CHANGES_REQUESTED** [review]

**Pull request merged**